tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundle.java

```diff
@@ -0,0 +1,273 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.server.core.portal;
+
+import com.liferay.ide.core.ILiferayConstants;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.StringPool;
+import com.liferay.ide.server.core.LiferayServerCore;
+import com.liferay.ide.server.util.LiferayPortalValueLoader;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import org.eclipse.core.runtime.IPath;
+import org.osgi.framework.Version;
+
+/**
+ * @author Simon Jiang
+ */
+public abstract class AbstractPortalBundle implements PortalBundle
+{
+    
+    private static final String CONFIG_TYPE_SERVER = "server"; //$NON-NLS-1$
+    private static final String CONFIG_TYPE_VERSION = "version"; //$NON-NLS-1$
+    private static final Version MANIFEST_VERSION_REQUIRED = ILiferayConstants.V700;
+
+
+    protected IPath autoDeployPath;
+    protected IPath liferayHome;
+    protected IPath modulesPath;
+    protected IPath bundlePath;
+    protected String version;
+    protected int jmxRemotePort;
+ 
+    public AbstractPortalBundle( IPath path )
+    {
+        if( path == null )
+        {
+            throw new IllegalArgumentException( "path cannot be null" );
+        }
+
+        this.bundlePath = path;
+        
+        this.liferayHome = bundlePath.append( ".." );
+        this.jmxRemotePort = detectJmxRemotePort();
+
+        this.autoDeployPath = this.liferayHome.append( "deploy" );
+
+        this.version = getPortalVersion( this.bundlePath, getPortalDir( bundlePath ) );
+
+        if( this.version != null && this.version.startsWith( "7" ) )
+        {
+            this.modulesPath = this.liferayHome.append( "data/osgi" );
```

Comment:
There is no need for a check... all portal bundles are going to be 7.0 or greater and they all have the osgi folder at liferayHome.append("osgi")